SEVERE: Error executing FreeMarker template

Hello
I have discovered an error while generating a NOTICE TEMPLATE report from scan-results file. I have used the ORT from a commit `a29c8c7237032c7e79daea2f763c508b58356c18`

```
18:31:16.925 [DefaultDispatcher-worker-2] INFO  org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor - Generating output file '/builds/sqsc/gopro/ort/reporter/NOTICE_default' using template id 'default'.
Feb 24, 2021 6:31:17 PM freemarker.log._JULLoggerFactory$JULLogger error
SEVERE: Error executing FreeMarker template
FreeMarker template error:
An error has occurred when reading existing sub-variable "licenseFiles"; see cause exception! The type of the containing value was: extended_hash+string (org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor$PackageModel wrapped into f.e.b.StringModel)
----
FTL stack trace ("~" means nesting-related):
	- Failed at: #list package.licenseFiles.files as l...  [in template "default.ftl" at line 78, column 1]
----
Java stack trace (for programmers):
----
freemarker.core._TemplateModelException: [... Exception message was already printed; see it above ...]
	at freemarker.ext.beans.BeanModel.get(BeanModel.java:185)
	at freemarker.core.Dot._eval(Dot.java:43)
	at freemarker.core.Expression.eval(Expression.java:101)
	at freemarker.core.Dot._eval(Dot.java:41)
	at freemarker.core.Expression.eval(Expression.java:101)
	at freemarker.core.IteratorBlock.acceptWithResult(IteratorBlock.java:99)
	at freemarker.core.IteratorBlock.accept(IteratorBlock.java:94)
	at freemarker.core.Environment.visit(Environment.java:334)
	at freemarker.core.Environment.visit(Environment.java:376)
	at freemarker.core.IteratorBlock$IterationContext.executedNestedContentForCollOrSeqListing(IteratorBlock.java:321)
	at freemarker.core.IteratorBlock$IterationContext.executeNestedContent(IteratorBlock.java:271)
	at freemarker.core.IteratorBlock$IterationContext.accept(IteratorBlock.java:244)
	at freemarker.core.Environment.visitIteratorBlock(Environment.java:644)
	at freemarker.core.IteratorBlock.acceptWithResult(IteratorBlock.java:108)
	at freemarker.core.IteratorBlock.accept(IteratorBlock.java:94)
	at freemarker.core.Environment.visit(Environment.java:334)
	at freemarker.core.Environment.visit(Environment.java:340)
	at freemarker.core.Environment.process(Environment.java:313)
	at freemarker.template.Template.process(Template.java:383)
	at org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor.processTemplatesInternal(FreemarkerTemplateProcessor.kt:149)
	at org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor.processTemplates(FreemarkerTemplateProcessor.kt:83)
	at org.ossreviewtoolkit.reporter.reporters.NoticeTemplateReporter.generateReport(NoticeTemplateReporter.kt:66)
	at org.ossreviewtoolkit.commands.ReporterCommand$run$$inlined$measureTimedValue$lambda$1$1.invokeSuspend(ReporterCommand.kt:266)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at freemarker.ext.beans.BeansWrapper.invokeMethod(BeansWrapper.java:1552)
	at freemarker.ext.beans.BeanModel.invokeThroughDescriptor(BeanModel.java:233)
	at freemarker.ext.beans.BeanModel.get(BeanModel.java:152)
	... 28 more
Caused by: java.lang.NullPointerException
	at org.ossreviewtoolkit.model.utils.FileArchiverKt.hash(FileArchiver.kt:152)
	at org.ossreviewtoolkit.model.utils.FileArchiverKt.getArchivePath(FileArchiver.kt:142)
	at org.ossreviewtoolkit.model.utils.FileArchiverKt.access$getArchivePath(FileArchiver.kt:1)
	at org.ossreviewtoolkit.model.utils.FileArchiver.unarchive(FileArchiver.kt:114)
	at org.ossreviewtoolkit.model.licenses.LicenseInfoResolver.createLicenseFileInfo(LicenseInfoResolver.kt:210)
	at org.ossreviewtoolkit.model.licenses.LicenseInfoResolver.resolveLicenseFiles(LicenseInfoResolver.kt:64)
	at org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor$PackageModel$licenseFiles$2.invoke(FreemarkerTemplateProcessor.kt:194)
	at org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor$PackageModel$licenseFiles$2.invoke(FreemarkerTemplateProcessor.kt:174)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor$PackageModel.getLicenseFiles(FreemarkerTemplateProcessor.kt)
	... 35 more
18:31:19.647 [main] ERROR org.ossreviewtoolkit.commands.ReporterCommand - Could not create 'NoticeTemplate' report in 0.207224318s: _TemplateModelException: An error has occurred when reading existing sub-variable "licenseFiles"; see cause exception! The type of the containing value was: extended_hash+string (org.ossreviewtoolkit.reporter.utils.FreemarkerTemplateProcessor$PackageModel wrapped into f.e.b.StringModel)
----
FTL stack trace ("~" means nesting-related):
	- Failed at: #list package.licenseFiles.files as l...  [in template "default.ftl" at line 78, column 1]
----
Caused by: InvocationTargetException: null
Caused by: NullPointerException: null
```

--- a/model/src/main/kotlin/utils/FileArchiver.kt
+++ b/model/src/main/kotlin/utils/FileArchiver.kt
@@ -68,8 +68,9 @@ class FileArchiver(
      * Return whether an archive corresponding to [provenance] exists.
      */
     fun hasArchive(provenance: Provenance): Boolean {
-        val archivePath = getArchivePath(provenance)
+        if (provenance.sourceArtifact == null && provenance.vcsInfo == null) return false
 
+        val archivePath = getArchivePath(provenance)
         return storage.exists(archivePath)
     }
 
@@ -77,6 +78,10 @@ class FileArchiver(
      * Archive all files in [directory] matching any of the configured [patterns] in the [storage].
      */
     fun archive(directory: File, provenance: Provenance) {
+        require(provenance.sourceArtifact != null || provenance.vcsInfo != null) {
+            "Unable to create an archive for unknown provenance."
+        }
+
         val zipFile = createTempFile(ORT_NAME, ".zip").toFile()
 
         val zipDuration = measureTime {
@@ -111,6 +116,8 @@ class FileArchiver(
      * Unarchive the archive corresponding to [provenance].
      */
     fun unarchive(directory: File, provenance: Provenance): Boolean {
+        if (provenance.sourceArtifact == null && provenance.vcsInfo == null) return false
+
         val archivePath = getArchivePath(provenance)
 
         return try {

